.eslintrc.js

```diff
@@ -16,8 +16,10 @@ module.exports = {
 		"no-undef": "error",
 		"no-extra-semi": "error",
 		"semi": "error",
+		"quotes": ["error", "double", { "avoidEscape": true }],
```

Comment:
The default is `"double"` and that's fine. Stay with the default. Remove this line.

Reply:
@sokra, if I remove this line `quotes` rule will not be enabled because in `eslint:recommended` is disabled and in  `plugin:node/recommended` is not present.

Also if I'm specify "double" and `avoidEscape` option to avoid errors in scenarios like:
```
"process.env.DEFINED_NESTED_KEY_STRING": '"string"',
```
https://github.com/webpack/webpack/blob/master/test/configCases/plugins/define-plugin/webpack.config.js#L25

Reply:
@marcalexiei This is not an issue. Prettier handles this for us.

Reply:
@ooflorent, I might be missing something, but let me go over this once again just to be sure.

### Without this rule setting
Using the template string without `${}` will *not* display a warning.
Running `pretty-files`, `lint`, `lint-files` and `fix` *won't* change \`\` to "". 

Prettier probably doesn't handle this scenario, since it is just a code formatter.
When it finds a template string, it simply leaves it as it is, regardless of any `${}` usage in it.

### With this rule setting
Eslint will actually check for `${}` usages inside template strings, and notify the developer if they're used improperly.

Also, strings will be fixed when running the `fix` script.

Reply:
from https://eslint.org/docs/rules/quotes

> `"avoidEscape": true` allows strings to use single-quotes or double-quotes so long as the string contains a quote that would have to be escaped otherwise

from https://eslint.org/docs/rules/no-template-curly-in-string

> This rule aims to warn when a regular string contains what looks like a template literal placeholder. It will warn when it finds a string containing the template literal place holder (`${something}`) that uses either `"` or `'` for the quotes.

---

In my opinion you are talking about `no-template-curly-in-string`, not about `quotes` with `avoidEscape`

Reply:
In this screen you see a template string without `${variable}` and error is not present (Code review in #6609).

<img width="1323" alt="no-error" src="https://user-images.githubusercontent.com/24919330/37361299-db27ad60-26f2-11e8-8995-9af77e351310.png">

With `quotes` rule set in `.eslintrc.json` an error is correctly generated:

<img width="1309" alt="error" src="https://user-images.githubusercontent.com/24919330/37361298-db0612ae-26f2-11e8-8c63-fac89dcf0e2e.png">

---

`avoidEscape` option has been added to avoid errors in scenario like string in `/test/Validation.test.js`:

```
' - configuration.context: The provided value "baz" is not an absolute path!'
```

<img width="1517" alt="avoid-escape" src="https://user-images.githubusercontent.com/24919330/37361297-dae2810e-26f2-11e8-8366-92fd0e8a9af4.png">

`no-template-curly-in-string` rule is not involved in this scenario.

I hope screenshots will speak better then me 😁.